mwclient/client.py

```diff
@@ -89,17 +88,40 @@ class Site:
     """
     api_limit = 500
 
-    def __init__(self, host, path='/w/', ext='.php', pool=None, retry_timeout=30,
-                 max_retries=25, wait_callback=lambda *x: None, clients_useragent=None,
-                 max_lag=3, compress=True, force_login=True, do_init=True, httpauth=None,
-                 reqs=None, consumer_token=None, consumer_secret=None, access_token=None,
-                 access_secret=None, client_certificate=None, custom_headers=None,
-                 scheme='https'):
+    def __init__(
+        self,
+        host: str,
+        path: str = '/w/',
+        ext: str = '.php',
+        pool: Optional[requests.Session] = None,
+        retry_timeout: int = 30,
+        max_retries: int = 25,
+        wait_callback: Callable[..., None] = lambda *x: None,
```

Comment:
I don't think `...` is correct here. The doc says "If a literal ellipsis ... is given as the argument list, it indicates that a callable with any arbitrary parameter list would be acceptable", but this callable has a specific expected signature. It's ultimately called (in `sleep.py`) as `self.callback(self, self.retries, self.args)`. `self.args` comes from a call to `Sleepers.make` and defaults to `None`, and all the calls to `self.sleepers.make` in `client.py` are just `self.sleepers.make()`, so `self.args` for the individual sleeper will always be `None`. So I think this callback must accept two args, the first being an `int` and the second being `None`. On the other hand, I don't think it has to *return* `None`, because its return value is ignored when it's called, so it could return anything. So...I think this should be `Callable[[int, None], Any]`. Or am I wrong? (this is a pretty odd interface, now I dig into it, I'd be surprised if anyone figured out how to use it...:>)

Reply:
I guess *technically* it should be something like `Callable[[int, Optional[Any]], Any]`...

Reply:
Yes, you're correct. I had just taken the signature from the old docstring. Actually, I think the correct type is `Callable[['Sleeper', int, Optional[Any]], Any]` since the `sleep.Sleeper` is passed as the first argument:

```python
class Sleeper:
  # ...
  def sleep(self, min_time=0):
    # ...
    self.callback(self, self.retries, self.args)
```

I've pushed an updated version.